addons/sale/models/sale.py

```diff
@@ -821,7 +857,7 @@ def _create_payment_transaction(self, vals):
             vals['acquirer_id'] = acquirer.id
 
         vals.update({
-            'amount': sum(self.mapped('amount_total')),
+            'amount': sum(self.mapped(lambda sol: sol.confirmation_amount if sol.has_to_be_down_paid() else sol.amount_total)),
```

Comment:
I think a list comprehension will be easier to understand here (and probably better for performances).